cbcontainers/communication/gateway/api_gateway.go

```diff
@@ -21,9 +20,34 @@ type ApiGateway struct {
 	client      *resty.Client
 }
 
-func NewApiGateway(account, cluster string, accessToken string, scheme, host string, port int, adapter string) *ApiGateway {
+func createClient(tlsInsecureSkipVerify bool, rootCAsBundle []byte) (*resty.Client, error) {
 	client := resty.New()
-	client.SetTLSClientConfig(&tls.Config{InsecureSkipVerify: true})
+	tlsConfig := tls.Config{
+		InsecureSkipVerify: tlsInsecureSkipVerify,
+	}
+
+	if len(rootCAsBundle) > 0 {
+		rootCAs := x509.NewCertPool()
+		ok := rootCAs.AppendCertsFromPEM(rootCAsBundle)
+		if !ok {
+			return nil, fmt.Errorf("failed to parse root certificates")
+		}
+
+		tlsConfig.RootCAs = rootCAs
+	}
```

Comment:
Should we output some form of warning when `InsecureSkipVerify` is true since that is a pretty big deal? 

Or at least when `InsecureSkipVerify` is true *and* `RootCAs` is also set, because I don't see how this makes sense